### PR TITLE
Desktop:  Add test coverage

### PR DESCRIPTION
**Pull request by rihardsgravis**
# Description

Add Unit Tests for:

- [x] React UI components
- [x] React UI views visual
- [x] Native library wrappers
- [x] SeedStore classes
- [x] End-to-end tests for:
  - [x] Onboarding existing seed 
  - [x] Onboarding new seed
  - [x] Send & receive

## Type of change

- Enhancement (a non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/desktop/__tests__/ui/components/list.test.jsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import { ListComponent as List } from 'ui/components/List';
+
```
Comment (rajivshah3) [inline src/desktop/__tests__/ui/components/list.test.jsx:5]: The snapshot test may fail when run in an environment that uses a different date format (e.g. MM/DD/YYYY or DD.MM.YYYY). We should consider mocking `moment` to resolve this (see https://medium.com/front-end-weekly/mocking-moment-js-in-jest-a-simple-one-line-solution-61259ffaaa2)

File: src/desktop/__tests__/ui/global/errorLog.test.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import { ErrorLogComponent as ErrorLog } from 'ui/global/ErrorLog';
+
```
Comment (rajivshah3) [inline src/desktop/__tests__/ui/global/errorLog.test.jsx:5]: `moment` may need to be mocked here as well

File: src/desktop/__tests__/e2e/Onboarding.test.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            // Seed intro
+            await page.waitForSelector('#to-seed-generate');
+            await page.click('#to-seed-generate');
+
+            // Seed generate
+            await new Promise((resolve) => setTimeout(resolve, 200));
+            for (let i = 1; i <= 10; i++) {
+                const selector = `.index__seed button:nth-child(${i})`;
+                await page.waitForSelector(selector);
+                await page.click(selector);
```
Comment (rajivshah3) [inline src/desktop/__tests__/e2e/Onboarding.test.js:67]: This one seemed to fail twice out of the eight times I ran it. Is there anything we can do to help reduce the flakiness or is this an issue with the component itself?


```
 FAIL  __tests__/e2e/Onboarding.test.js (9.132s)
  Trinity desktop end-to-end
    ✓ Trinity onboarding existing seed (3496ms)
    ✕ Trinity onboarding new seed (3061ms)

  ● Trinity desktop end-to-end › Trinity onboarding new seed

    Node is either not visible or not an HTMLElement

      65 |                 const selector = `.index__seed button:nth-child(${i})`;
      66 |                 await page.waitForSelector(selector);
    > 67 |                 await page.click(selector);
         |                            ^
      68 |             }
      69 |             await page.click('#to-account-name');
      70 | 

      at ElementHandle._clickablePoint (node_modules/puppeteer/lib/JSHandle.js:199:13)
        -- ASYNC --
      at ElementHandle.<anonymous> (node_modules/puppeteer/lib/helper.js:110:27)
      at DOMWorld.click (node_modules/puppeteer/lib/DOMWorld.js:367:18)
        -- ASYNC --
      at Frame.<anonymous> (node_modules/puppeteer/lib/helper.js:110:27)
      at Page.click (node_modules/puppeteer/lib/Page.js:988:29)
      at Object.click (__tests__/e2e/Onboarding.test.js:67:28)
```

Comment (rihardsgravis) [reply src/desktop/__tests__/e2e/Onboarding.test.js:67]: It's Puppeteer's new page initialisation that causes the random delay. Will look into on how to reduce it.

File: src/desktop/__tests__/e2e/Onboarding.test.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+            // Seed intro
+            await page.waitForSelector('#to-seed-verify');
+            await page.click('#to-seed-verify');
+
+            // Seed input
+            await page.waitForSelector('.input__editable > div');
+            await page.focus('.input__editable > div');
+            await page.keyboard.type(
+                '999999999999999999999999999999999999999999999999999999999999999999999999999999999',
```
Comment (rajivshah3) [inline src/desktop/__tests__/e2e/Onboarding.test.js]:
```suggestion
                '9'.repeat(81),
```

For easier readability

File: src/desktop/__tests__/e2e/Onboarding.test.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+            // Seed save
+            await page.waitForSelector('#to-seed-verify');
+            await page.click('#to-seed-verify');
+
+            // Seed verify
+            await page.waitForSelector('.input__editable > div');
+            await page.focus('.input__editable > div');
+            await page.keyboard.type(
+                '999999999999999999999999999999999999999999999999999999999999999999999999999999999',
```
Comment (rajivshah3) [inline src/desktop/__tests__/e2e/Onboarding.test.js]:
```suggestion
                '9'.repeat(81),
```

File: src/desktop/__tests__/libs/seedStore/keychain.test.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    });
+
+    test('Get seed in trytes form', async () => {
+        const seedVault = await new Keychain(passwordMock, 'Foo');
+
+        const seed = await seedVault.getSeed(true);
+        expect(seed).toEqual(seedTrytesMock);
+    });
+
+    test('Generate address', async () => {
```
Comment (rajivshah3) [inline src/desktop/__tests__/libs/seedStore/keychain.test.js]:
```suggestion
    test('Generate address and call garbage collector', async () => {
```

**Comment by rajivshah3** [comment]
Buildkite CI is failing:

```

npm ERR! code ENOLOCAL
npm ERR! Could not install from "node_modules/kdbxweb/pako@github:keeweb/pako#653c0b00d8941c89d09ed4546d2179001ec44efc" as it does not contain a package.json file.

```
Will investigate this, I can't repro locally though 🤔

File: src/desktop/package.json [hunk 7]
```diff
@@ -21,47 +20,53 @@
...
     "mini-css-extract-plugin": "0.6.0",
     "puppeteer": "1.16.0",
     "react-dev-utils": "9.0.1",
+    "regenerator-runtime": "^0.13.2",
     "rimraf": "2.6.3",
     "sass-loader": "7.1.0",
     "style-loader": "0.23.1",
     "url-loader": "1.1.2",
-    "webpack": "4.31.0",
+    "usb": "1.5.0",
```
Comment (rajivshah3) [inline src/desktop/package.json]: Is this intentional?

```
npm WARN The package usb is included as both a dev and production dependency.
```

Comment (rihardsgravis) [reply src/desktop/package.json]: No, it was a typo 👍

**Pull request closed without merging**